### DISCUSSION TITLE
How to tell if a property is explicitly set or is inherited

### DISCUSSION THREAD

**Discussion by SimonCropp** [AvaloniaUI/Avalonia#12626] [Q&A]
so i have a TextBox 

```
 <TextBox Name="FirstOperandInput" Text="{Binding FirstOperand, Mode=TwoWay}" Watermark="Operand 1" />
```

When looking at that TextBox in memory, how do i tell if a given property is explicitly set. For example is i look at the `SelectionBrush` property i get `#ff0078d7` which is inherited from some default style. So is there some way using `TextBox.SelectionBrushProperty` and a `TextBox` instance to work out that the `SelectionBrush` property is inherited?

**Comment by wieslawsoltes** [comment]
Try using DevTools, it has applied styles panel that shows which properties are applied.

**Reply by SimonCropp** [reply]
@wieslawsoltes thanks. but i need to know how to do it programmatically

**Reply by wieslawsoltes** [reply]
You can check diagnostics code: https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Diagnostics/Diagnostics/ViewModels/ControlDetailsViewModel.cs

**Reply by wieslawsoltes** [reply]
![image](https://github.com/AvaloniaUI/Avalonia/assets/2297442/4303fce8-8894-4e0e-b511-fefddb74120a)
